src/check/canonicalize.zig

```diff
@@ -203,6 +243,66 @@ const PendingValueDef = union(enum) {
     pub const List = collections.SafeList(@This());
 };
 
+fn canonicalize_decl(
+    decl: parse.IR.NodeStore.Statement.Decl,
+    parse_ir: *parse.IR,
+    ir: *IR,
+    scope: *Scope,
+) void {
+    _ = canonicalize_expr(decl.body, parse_ir, ir, scope);
+    canonicalize_pattern(decl.pattern, parse_ir, ir, scope);
+}
+
+fn canonicalize_expr(
+    expr_idx: parse.IR.NodeStore.ExprIdx,
+    parse_ir: *parse.IR,
+    ir: *IR,
+    scope: *Scope,
+) ?IR.Expr.Idx {
+    const expr = parse_ir.store.getExpr(expr_idx);
+    switch (expr) {
+        .apply => |e| {
+            _ = canonicalize_expr(e.@"fn", parse_ir, ir, scope);
+            const args_slice = parse_ir.store.exprSlice(e.args);
+            for (args_slice) |arg| {
+                _ = canonicalize_expr(arg, parse_ir, ir, scope);
+            }
+        },
+        .ident => |e| {
+            if (parse_ir.tokens.resolveIdentifier(e.token)) |ident| {
+                switch (scope.levels.lookup(.ident, ident)) {
+                    .InScope => {
+                        return ir.exprs.append(ir.env.gpa, .{ .@"var" = .{
```

Comment:
I'm not sure if this was the original intention, but I assumed "`var`" was referring to the new `var` keyword for reassignable defs.

Personally I like the name `lookup` for this because it's an expression that's looking up something by name.